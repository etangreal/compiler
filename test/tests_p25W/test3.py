if ( True ):
	print 1

	if ( False ):
		print 2

		x = 1
		while x < 3:
			print 100
			x = x + 1

			while False:
				print 200

				while True:
					print 1000

					if True:
						print 300

					elif False:
						print 20

					elif True:
						print 33
						print 30

					else:
						print 400

	elif False:
		print 1

	elif True:
		print 12
		print 120

elif( False ):
	if ( False ):
		print 2

		y = 5
		while y > 3:
			print 100
			y = y - 1
else:
	print 1