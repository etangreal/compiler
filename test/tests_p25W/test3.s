	.section	__TEXT,__text,regular,pure_instructions
	.globl		_main
	.align		4, 0x90
_main:
	pushl	%ebp
	movl	%esp,%ebp
	subl	$632,%esp
	movl	$1,%eax
	movl	%eax,-4(%ebp)
	movl	-4(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_bool
	movl	%eax,-8(%ebp)
	movl	-8(%ebp),%eax
	movl	%eax,-12(%ebp)
	movl	-12(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-16(%ebp)
	movl	-16(%ebp),%eax
	cmpl	 $1,%eax
	jne	NEXT_IF_t2
	movl	$1,%eax
	movl	%eax,-20(%ebp)
	movl	-20(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-24(%ebp)
	movl	-24(%ebp),%eax
	movl	%eax,-28(%ebp)
	movl	-28(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-32(%ebp)
	movl	-32(%ebp),%eax
	movl	%eax,(%esp)
	call	_print_int_nl
	movl	$0,%eax
	movl	%eax,-36(%ebp)
	movl	-36(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_bool
	movl	%eax,-40(%ebp)
	movl	-40(%ebp),%eax
	movl	%eax,-44(%ebp)
	movl	-44(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-48(%ebp)
	movl	-48(%ebp),%eax
	cmpl	 $1,%eax
	jne	NEXT_IF_t12
	movl	$2,%eax
	movl	%eax,-52(%ebp)
	movl	-52(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-56(%ebp)
	movl	-56(%ebp),%eax
	movl	%eax,-60(%ebp)
	movl	-60(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-64(%ebp)
	movl	-64(%ebp),%eax
	movl	%eax,(%esp)
	call	_print_int_nl
	movl	$1,%eax
	movl	%eax,-68(%ebp)
	movl	-68(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-72(%ebp)
	movl	-72(%ebp),%eax
	movl	%eax,-76(%ebp)
	jmp	BOT_WHILE_t23
TOP_WHILE_t23:
	movl	-80(%ebp),%eax
	cmpl	 $0,%eax
	je	END_WHILE_t23
	movl	$100,%eax
	movl	%eax,-84(%ebp)
	movl	-84(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-88(%ebp)
	movl	-88(%ebp),%eax
	movl	%eax,-92(%ebp)
	movl	-92(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-96(%ebp)
	movl	-96(%ebp),%eax
	movl	%eax,(%esp)
	call	_print_int_nl
	movl	-76(%ebp),%eax
	movl	%eax,-100(%ebp)
	movl	-100(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-104(%ebp)
	movl	$1,%eax
	movl	%eax,-108(%ebp)
	movl	-108(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-112(%ebp)
	movl	-112(%ebp),%eax
	movl	%eax,-116(%ebp)
	movl	-116(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-120(%ebp)
	movl	-104(%ebp), %eax
	movl	-120(%ebp), %ecx
	movl	%eax, -124(%ebp)
	addl	%ecx,-124(%ebp)
	movl	-124(%ebp),%eax
	movl	%eax,-128(%ebp)
	movl	-128(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-132(%ebp)
	movl	-132(%ebp),%eax
	movl	%eax,-76(%ebp)
	jmp	BOT_WHILE_t48
TOP_WHILE_t48:
	movl	-136(%ebp),%eax
	cmpl	 $0,%eax
	je	END_WHILE_t48
	movl	$200,%eax
	movl	%eax,-140(%ebp)
	movl	-140(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-144(%ebp)
	movl	-144(%ebp),%eax
	movl	%eax,-148(%ebp)
	movl	-148(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-152(%ebp)
	movl	-152(%ebp),%eax
	movl	%eax,(%esp)
	call	_print_int_nl
	jmp	BOT_WHILE_t57
TOP_WHILE_t57:
	movl	-156(%ebp),%eax
	cmpl	 $0,%eax
	je	END_WHILE_t57
	movl	$1000,%eax
	movl	%eax,-160(%ebp)
	movl	-160(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-164(%ebp)
	movl	-164(%ebp),%eax
	movl	%eax,-168(%ebp)
	movl	-168(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-172(%ebp)
	movl	-172(%ebp),%eax
	movl	%eax,(%esp)
	call	_print_int_nl
	movl	$1,%eax
	movl	%eax,-176(%ebp)
	movl	-176(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_bool
	movl	%eax,-180(%ebp)
	movl	-180(%ebp),%eax
	movl	%eax,-184(%ebp)
	movl	-184(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-188(%ebp)
	movl	-188(%ebp),%eax
	cmpl	 $1,%eax
	jne	NEXT_IF_t67
	movl	$300,%eax
	movl	%eax,-192(%ebp)
	movl	-192(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-196(%ebp)
	movl	-196(%ebp),%eax
	movl	%eax,-200(%ebp)
	movl	-200(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-204(%ebp)
	movl	-204(%ebp),%eax
	movl	%eax,(%esp)
	call	_print_int_nl
	jmp	END_IF_t66
NEXT_IF_t67:
	movl	$0,%eax
	movl	%eax,-208(%ebp)
	movl	-208(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_bool
	movl	%eax,-212(%ebp)
	movl	-212(%ebp),%eax
	movl	%eax,-216(%ebp)
	movl	-216(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-220(%ebp)
	movl	-220(%ebp),%eax
	cmpl	 $1,%eax
	jne	NEXT_IF_t76
	movl	$20,%eax
	movl	%eax,-224(%ebp)
	movl	-224(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-228(%ebp)
	movl	-228(%ebp),%eax
	movl	%eax,-232(%ebp)
	movl	-232(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-236(%ebp)
	movl	-236(%ebp),%eax
	movl	%eax,(%esp)
	call	_print_int_nl
	jmp	END_IF_t66
NEXT_IF_t76:
	movl	$1,%eax
	movl	%eax,-240(%ebp)
	movl	-240(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_bool
	movl	%eax,-244(%ebp)
	movl	-244(%ebp),%eax
	movl	%eax,-248(%ebp)
	movl	-248(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-252(%ebp)
	movl	-252(%ebp),%eax
	cmpl	 $1,%eax
	jne	NEXT_IF_t85
	movl	$33,%eax
	movl	%eax,-256(%ebp)
	movl	-256(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-260(%ebp)
	movl	-260(%ebp),%eax
	movl	%eax,-264(%ebp)
	movl	-264(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-268(%ebp)
	movl	-268(%ebp),%eax
	movl	%eax,(%esp)
	call	_print_int_nl
	movl	$30,%eax
	movl	%eax,-272(%ebp)
	movl	-272(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-276(%ebp)
	movl	-276(%ebp),%eax
	movl	%eax,-280(%ebp)
	movl	-280(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-284(%ebp)
	movl	-284(%ebp),%eax
	movl	%eax,(%esp)
	call	_print_int_nl
	jmp	END_IF_t66
NEXT_IF_t85:
	movl	$400,%eax
	movl	%eax,-288(%ebp)
	movl	-288(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-292(%ebp)
	movl	-292(%ebp),%eax
	movl	%eax,-296(%ebp)
	movl	-296(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-300(%ebp)
	movl	-300(%ebp),%eax
	movl	%eax,(%esp)
	call	_print_int_nl
	jmp	END_IF_t66
END_IF_t66:
BOT_WHILE_t57:
	movl	$1,%eax
	movl	%eax,-304(%ebp)
	movl	-304(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_bool
	movl	%eax,-308(%ebp)
	movl	-308(%ebp),%eax
	movl	%eax,-312(%ebp)
	movl	-312(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-156(%ebp)
	jmp	TOP_WHILE_t57
	jmp	END_WHILE_t57
END_WHILE_t57:
BOT_WHILE_t48:
	movl	$0,%eax
	movl	%eax,-316(%ebp)
	movl	-316(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_bool
	movl	%eax,-320(%ebp)
	movl	-320(%ebp),%eax
	movl	%eax,-324(%ebp)
	movl	-324(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-136(%ebp)
	jmp	TOP_WHILE_t48
	jmp	END_WHILE_t48
END_WHILE_t48:
BOT_WHILE_t23:
	movl	-76(%ebp),%eax
	movl	%eax,-328(%ebp)
	movl	-328(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-332(%ebp)
	movl	$3,%eax
	movl	%eax,-336(%ebp)
	movl	-336(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-340(%ebp)
	movl	-340(%ebp),%eax
	movl	%eax,-344(%ebp)
	movl	-344(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-348(%ebp)
	movl	$0,-352(%ebp)
	movl	-332(%ebp),%eax
	cmpl	-348(%ebp),%eax
	jl	LA1
	jmp	LB1
LA1:
	movl	$1,-352(%ebp)
LB1:
	movl	-352(%ebp),%eax
	movl	%eax,-356(%ebp)
	movl	-356(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_bool
	movl	%eax,-360(%ebp)
	movl	-360(%ebp),%eax
	movl	%eax,-364(%ebp)
	movl	-364(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-80(%ebp)
	jmp	TOP_WHILE_t23
	jmp	END_WHILE_t23
END_WHILE_t23:
	jmp	END_IF_t11
NEXT_IF_t12:
	movl	$0,%eax
	movl	%eax,-368(%ebp)
	movl	-368(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_bool
	movl	%eax,-372(%ebp)
	movl	-372(%ebp),%eax
	movl	%eax,-376(%ebp)
	movl	-376(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-380(%ebp)
	movl	-380(%ebp),%eax
	cmpl	 $1,%eax
	jne	NEXT_IF_t102
	movl	$1,%eax
	movl	%eax,-384(%ebp)
	movl	-384(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-388(%ebp)
	movl	-388(%ebp),%eax
	movl	%eax,-392(%ebp)
	movl	-392(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-396(%ebp)
	movl	-396(%ebp),%eax
	movl	%eax,(%esp)
	call	_print_int_nl
	jmp	END_IF_t11
NEXT_IF_t102:
	movl	$1,%eax
	movl	%eax,-400(%ebp)
	movl	-400(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_bool
	movl	%eax,-404(%ebp)
	movl	-404(%ebp),%eax
	movl	%eax,-408(%ebp)
	movl	-408(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-412(%ebp)
	movl	-412(%ebp),%eax
	cmpl	 $1,%eax
	jne	NEXT_IF_t111
	movl	$12,%eax
	movl	%eax,-416(%ebp)
	movl	-416(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-420(%ebp)
	movl	-420(%ebp),%eax
	movl	%eax,-424(%ebp)
	movl	-424(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-428(%ebp)
	movl	-428(%ebp),%eax
	movl	%eax,(%esp)
	call	_print_int_nl
	movl	$120,%eax
	movl	%eax,-432(%ebp)
	movl	-432(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-436(%ebp)
	movl	-436(%ebp),%eax
	movl	%eax,-440(%ebp)
	movl	-440(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-444(%ebp)
	movl	-444(%ebp),%eax
	movl	%eax,(%esp)
	call	_print_int_nl
	jmp	END_IF_t11
NEXT_IF_t111:
	jmp	END_IF_t11
END_IF_t11:
	jmp	END_IF_t1
NEXT_IF_t2:
	movl	$0,%eax
	movl	%eax,-448(%ebp)
	movl	-448(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_bool
	movl	%eax,-452(%ebp)
	movl	-452(%ebp),%eax
	movl	%eax,-456(%ebp)
	movl	-456(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-460(%ebp)
	movl	-460(%ebp),%eax
	cmpl	 $1,%eax
	jne	NEXT_IF_t124
	movl	$0,%eax
	movl	%eax,-464(%ebp)
	movl	-464(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_bool
	movl	%eax,-468(%ebp)
	movl	-468(%ebp),%eax
	movl	%eax,-472(%ebp)
	movl	-472(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-476(%ebp)
	movl	-476(%ebp),%eax
	cmpl	 $1,%eax
	jne	NEXT_IF_t130
	movl	$2,%eax
	movl	%eax,-480(%ebp)
	movl	-480(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-484(%ebp)
	movl	-484(%ebp),%eax
	movl	%eax,-488(%ebp)
	movl	-488(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-492(%ebp)
	movl	-492(%ebp),%eax
	movl	%eax,(%esp)
	call	_print_int_nl
	movl	$5,%eax
	movl	%eax,-496(%ebp)
	movl	-496(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-500(%ebp)
	movl	-500(%ebp),%eax
	movl	%eax,-504(%ebp)
	jmp	BOT_WHILE_t141
TOP_WHILE_t141:
	movl	-508(%ebp),%eax
	cmpl	 $0,%eax
	je	END_WHILE_t141
	movl	$100,%eax
	movl	%eax,-512(%ebp)
	movl	-512(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-516(%ebp)
	movl	-516(%ebp),%eax
	movl	%eax,-520(%ebp)
	movl	-520(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-524(%ebp)
	movl	-524(%ebp),%eax
	movl	%eax,(%esp)
	call	_print_int_nl
	movl	-504(%ebp),%eax
	movl	%eax,-528(%ebp)
	movl	-528(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-532(%ebp)
	movl	$1,%eax
	movl	%eax,-536(%ebp)
	movl	-536(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-540(%ebp)
	movl	-540(%ebp),%eax
	movl	%eax,-544(%ebp)
	movl	-544(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-548(%ebp)
	movl	-532(%ebp), %eax
	movl	%eax, -552(%ebp)
	movl	-548(%ebp), %eax
	subl	%eax,-552(%ebp)
	movl	-552(%ebp),%eax
	movl	%eax,-556(%ebp)
	movl	-556(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-560(%ebp)
	movl	-560(%ebp),%eax
	movl	%eax,-504(%ebp)
BOT_WHILE_t141:
	movl	-504(%ebp),%eax
	movl	%eax,-564(%ebp)
	movl	-564(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-568(%ebp)
	movl	$3,%eax
	movl	%eax,-572(%ebp)
	movl	-572(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-576(%ebp)
	movl	-576(%ebp),%eax
	movl	%eax,-580(%ebp)
	movl	-580(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-584(%ebp)
	movl	$0,-588(%ebp)
	movl	-568(%ebp),%eax
	cmpl	-584(%ebp),%eax
	jg	LA2
	jmp	LB2
LA2:
	movl	$1,-588(%ebp)
LB2:
	movl	-588(%ebp),%eax
	movl	%eax,-592(%ebp)
	movl	-592(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_bool
	movl	%eax,-596(%ebp)
	movl	-596(%ebp),%eax
	movl	%eax,-600(%ebp)
	movl	-600(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-508(%ebp)
	jmp	TOP_WHILE_t141
	jmp	END_WHILE_t141
END_WHILE_t141:
	jmp	END_IF_t129
NEXT_IF_t130:
	jmp	END_IF_t129
END_IF_t129:
	jmp	END_IF_t1
NEXT_IF_t124:
	movl	$1,%eax
	movl	%eax,-604(%ebp)
	movl	-604(%ebp),%eax
	movl	%eax,(%esp)
	call	_create_int
	movl	%eax,-608(%ebp)
	movl	-608(%ebp),%eax
	movl	%eax,-612(%ebp)
	movl	-612(%ebp),%eax
	movl	%eax,(%esp)
	call	_project_int
	movl	%eax,-616(%ebp)
	movl	-616(%ebp),%eax
	movl	%eax,(%esp)
	call	_print_int_nl
	jmp	END_IF_t1
END_IF_t1:
	movl	$0, %eax
	leave
	ret

