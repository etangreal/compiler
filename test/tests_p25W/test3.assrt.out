1
100
120